### PR TITLE
Fixed loading static files from FS or Jars

### PR DESCRIPTION
**Pull request by KofTwentyTwo**
Fixed the Javalin Static File feature, allowing it to serve static files from the FS or from Jars. 

In development (IDE), the setup is to always use the FS.   However, Once packaged and pushed to an environment, it would attempt to serve the files from the FS still, but they would be part of the jar. 

Javalin supports this, but it must be configured on a per "hostedDirectory" setup (as either EXTERNAL (FS), or CLASSPATH (jars)). 

This change adds a system property that can be set to configure Javalin to use one approach or the other for all static files.  (Note: other approaches were considered, however no valid use-cases really exist right now for having this configured at a more granular level).  

Also note, for testing, the test case for this was broken out into a standalone class, to handle the need to a higher level teardown between tests (ie, new classloader to null out the static nature of how we load and eval system properties)

### CODE REVIEW

File: qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java [hunk 1]
```diff
@@ -67,6 +65,24 @@ public SimpleFileSystemDirectoryRouter(String hostedPath, String fileSystemPath)
...
+         String propertyName  = "qqq.javalin.enableStaticFilesFromJar";  // TODO: make a more general way to handle properties like this system-wide via a central config class
+         String propertyValue = System.getProperty(propertyName, "");
+         if(propertyValue.equals("true"))
+         {
+            loadStaticFilesFromJar = true;
+         }
+      }
+      catch(Exception e)
+      {
+         e.printStackTrace();
```
Comment (darinkelkhoff) [inline qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java:84]: better to log.warn here, ala
```
LOG.warn("Error checking enableStaticFilesFromJar property", e);
```

File: qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java [hunk 2]
```diff
@@ -67,6 +65,24 @@ public SimpleFileSystemDirectoryRouter(String hostedPath, String fileSystemPath)
...
       this.hostedPath = hostedPath;
       this.fileSystemPath = fileSystemPath;
+
+      ///////////////////////////////////////////////////////////////////////////////////////////////////////
+      // read the property to see if we should load static files from the jar file or from the file system //
+      // Javan only supports loading via one method per path, so its a choice of one or the other...       //
+      ///////////////////////////////////////////////////////////////////////////////////////////////////////
+      try
+      {
+         String propertyName  = "qqq.javalin.enableStaticFilesFromJar";  // TODO: make a more general way to handle properties like this system-wide via a central config class
```
Comment (darinkelkhoff) [inline qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java:75]: would be nice to put this in, e.g., `public static final String ENABLE_STATIC_FILES_FROM_JAR_PROPERTY_NAME`, so your tests wouldn't have to have the string copy-pasted

File: qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java [hunk 3]
```diff
@@ -98,25 +114,41 @@ public void setQInstance(QInstance qInstance)
...
-      staticFileConfig.location = Location.EXTERNAL;
+      /// /////////////////////////////////////////////////////////////////////////////////////
+      // Handle loading static files from the jar OR the filesystem based on system property //
+      /// /////////////////////////////////////////////////////////////////////////////////////
+      if(SimpleFileSystemDirectoryRouter.loadStaticFilesFromJar)
+      {
+         staticFileConfig.directory = fileSystemPath;
+         staticFileConfig.hostedPath = hostedPath;
+         staticFileConfig.location = Location.CLASSPATH;
+         LOG.info("Static File Config : hostedPath [" + hostedPath + "] : directory [" + staticFileConfig.directory + "] : location [CLASSPATH]");
```
Comment (darinkelkhoff) [inline qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java:131]: nothing wrong with these log statements - but - as a convention of the codebase, we tend to put variable values in what we call "logPairs" - objects that QLogger knows to convert into json name:value pairs, e.g., for creating structured JSON logs that can be parsed as data in loggly.  

Using log pairs, these lines would look like:
```
         LOG.info("Static File Config", logPair("hostedPath", hostedPath), logPair("directory", staticFileConfig.directory), logPair("location", "CLASSPATH"));
```
(where `logPair` is a method that we always do a static-import for; it is a short-hand for the `new LogPair()` constructor)

this would then write a log output line in the style of:
```
{"message": "Static File Config", "hostedPath": "/foo", "directory": "/bar", "location": "CLASSPATH"}
```

it's actually a little less nice i guess for a human... but does have the benefit of the loggly parsing.  

i suppose, we could see an option in the future, in the logger, to NOT output json - in which case, log pairs could be printed as name=[value] style, which might be a preference for some (or for local dev vs. an env where pushing to a loggly)

anyway, passing the data to the logger as logPairs would give us the flexibility to do it either way - treating the log data as data instead of just a string.

**Review by darinkelkhoff — COMMENTED** [review]
looks good, though a few comments on conventions i suppose.

ooh, looks like you already merged to dev?

**Comment by darinkelkhoff** [comment]
@KofTwentyTwo do you get this  comment?

File: qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java (hunk 1 above)
Comment (KofTwentyTwo) [reply qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java:84]: ya, I agree for sure, was just copying existing examples in the code base.  But 100% agree and will update.

File: qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java (hunk 2 above)
Comment (KofTwentyTwo) [reply qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java:75]: Agree, but again, following existing examples in the codebase.   So, trying not to "make things better" when there is already a standard.   And you know me, I HATE random strings that make things work littered in all the places.

File: qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java (hunk 3 above)
Comment (KofTwentyTwo) [reply qqq-middleware-javalin/src/main/java/com/kingsrook/qqq/middleware/javalin/routeproviders/SimpleFileSystemDirectoryRouter.java:131]: nice and will do. 

Are there any good conventions in place on when we use DEBUG vs INFO vs?
